include/cadmium/engine/pdevs_simulator.hpp

```diff
@@ -216,20 +163,124 @@ namespace cadmium {
                     }
                 }
 
-                auto log_state = [](const typename model_type::state_type& s) -> std::string {
-                    std::ostringstream oss;
-                    oss << "State for model ";
-                    oss << boost::typeindex::type_id<model_type>().pretty_name();
-                    oss << " is ";
-                    oss << s;
-                    return oss.str();
-                };
-
-                LOGGER::template log<cadmium::logger::logger_state, decltype(log_state), const typename model_type::state_type&>(log_state, _model.state);
+                LOGGER::template log<cadmium::logger::logger_state, decltype(formatter::log_state), const typename model_type::state_type&>(formatter::log_state, _model.state);
             }
     //TODO: use enable_if functions to give access to read state and messages in debug mode
         };
 
+
+        template<template<typename T> class MODEL, typename TIME, typename LOGGER>
+        class dynamic_simulator {
```

Comment:
Maybe this needs to be derived from a "simulator" class that has all abstract methods so you can make a vector<unique_ptr<simulator>> for submodels of the coordinator.

Reply:
Some dynamic_simulator methods has the TIME type in their declaration, thus, the abstract base class where to derive dynamic_simulator is also a template TIME class, i think this is not a problem because all the pointer will use the same TIME type at the end, this is not the MODEL type.

Reply:
I've called the abstract class "abstract_dynamic_simulator"